packages/treasury/test/swingsetTests/test-treasury.js

```diff
@@ -0,0 +1,68 @@
+/* global __dirname */
+
+// @ts-check
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import '@agoric/install-ses';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import test from 'ava';
+import { buildVatController, buildKernelBundles } from '@agoric/swingset-vat';
+import bundleSource from '@agoric/bundle-source';
+
+import liquidateBundle from '../../bundles/bundle-liquidateMinimum';
+import autoswapBundle from '../../bundles/bundle-multipoolAutoswap';
+import stablecoinBundle from '../../bundles/bundle-stablecoinMachine';
+
+test.before(async t => {
+  const kernelBundles = await buildKernelBundles();
+
+  const vats = {};
+  await Promise.all(
+    ['alice', 'zoe', 'priceAuthority', 'owner'].map(async name => {
+      const source = `${__dirname}/vat-${name}.js`;
+      const bundle = await bundleSource(source);
+      vats[name] = { bundle };
+    }),
+  );
+
+  const nameToBundle = [
+    ['liquidateMinimum', liquidateBundle],
+    ['autoswap', autoswapBundle],
+    ['treasury', stablecoinBundle],
+  ];
+  const contractBundles = {};
+  await Promise.all(
+    nameToBundle.map(async ([name, bundle]) => {
+      contractBundles[name] = bundle;
+    }),
+  );
+
+  const bootstrapSource = `${__dirname}/bootstrap.js`;
+  vats.bootstrap = {
+    bundle: await bundleSource(bootstrapSource),
```

Comment:
can this be in the previous Promise.all?  Also, this `await` needs to move to the top level.